kinematics_base_test/src/test_kinematics_plugin.cpp

```diff
@@ -483,23 +520,121 @@ TEST(IKFastPlugin, getIKMultipleSolutions)
       EXPECT_NEAR(poses[0].orientation.z, new_poses[0].orientation.z, IK_NEAR);
       EXPECT_NEAR(poses[0].orientation.w, new_poses[0].orientation.w, IK_NEAR);
     }
+  }
 
+  ROS_INFO_STREAM("Success Rate: "
+                  << (double)success / kinematics_test.num_ik_multiple_tests_);
+  EXPECT_GT(success, 0.99 * kinematics_test.num_ik_multiple_tests_)
+      << "A total of " << num_ik_solutions << " ik solutions were found out of "
+      << kinematics_test.num_ik_multiple_tests_ << " tests.";
+  ROS_INFO_STREAM(
+      "Elapsed time: " << (ros::WallTime::now() - start_time).toSec());
+}
 
-  }
+TEST(IKFastPlugin, getnearestIKSolution) {
+  rdf_loader::RDFLoader rdf_loader_(ROBOT_DESCRIPTION_PARAM);
+  robot_model::RobotModelPtr kinematic_model;
+  const boost::shared_ptr<srdf::Model> &srdf_model = rdf_loader_.getSRDF();
+  const boost::shared_ptr<urdf::ModelInterface> &urdf_model =
+      rdf_loader_.getURDF();
+  kinematic_model.reset(new robot_model::RobotModel(urdf_model, srdf_model));
+  robot_model::JointModelGroup *joint_model_group =
+      kinematic_model->getJointModelGroup(
+          kinematics_test.kinematics_solver_->getGroupName());
 
+  // Test inverse kinematics
+  std::vector<double> seed, fk_values;
+  std::vector<std::vector<double>> solutions;
+  double timeout = 5.0;
+  kinematics::KinematicsQueryOptions options;
+  kinematics::KinematicsResult result;
 
-  ROS_INFO_STREAM("Success Rate: "<<(double)success/kinematics_test.num_ik_multiple_tests_);
-  EXPECT_GT(success , 0.99 * kinematics_test.num_ik_multiple_tests_)<<"A total of "<<num_ik_solutions <<" ik solutions were found out of "
-      <<kinematics_test.num_ik_multiple_tests_<<" tests.";
-  ROS_INFO_STREAM("Elapsed time: "<< (ros::WallTime::now()-start_time).toSec());
-}
+  // getIK
+  moveit_msgs::MoveItErrorCodes error_code;
+  std::vector<double> solution_getIK;
+  solution_getIK.resize(
+      kinematics_test.kinematics_solver_->getJointNames().size(), 0.0);
 
+  std::vector<std::string> fk_names;
+  fk_names.push_back(kinematics_test.kinematics_solver_->getTipFrame());
+  robot_state::RobotState kinematic_state(kinematic_model);
 
+  double min = -3.14;
```

Comment:
I did that on purpose. Even when the seed is out of bound, our solver should be able to give the closest solution. I actually want to modify the test which can seed in some other convention. I mean instead of seed being in range -pi to pi. If someone is working in 0 to pi. Still the solver should be able to give the solution closest to that seed no matter what convention we are following. Do you think it is a good idea?
I have tried incorporating this call in ros-planning/moveit/pull/537.